iothub/device/tests/MqttTransportHandlerTests.cs

```diff
@@ -138,6 +138,26 @@ MqttTransportHandler CreateTransportHandlerWithMockChannel(out IChannel channel,
             return transport;
         }
 
+        MqttTransportHandler CreateTransportHandlerWithRealChannel(out IChannel channel, string connectionString = DummyConnectionString)
```

Comment:
Please add an E2E test. While the UT is good to have, we need to have the scenario handled.
My recommendation would be to use TcpListener and attempt a channel connect.